tls/s2n_handshake_type.c

```diff
@@ -57,6 +57,7 @@ S2N_RESULT s2n_handshake_type_set_tls13_flag(struct s2n_connection *conn, s2n_tl
     RESULT_ENSURE_REF(conn);
     RESULT_ENSURE(s2n_connection_get_protocol_version(conn) >= S2N_TLS13, S2N_ERR_HANDSHAKE_STATE);
     conn->handshake.handshake_type |= flag;
+    RESULT_GUARD(s2n_conn_choose_state_machine(conn, S2N_TLS13));
```

Comment:
Oh do we need this for s2n_handshake_type_set_tls12_flag too? Do we set any TLS1.2 flags early?

Reply:
Technically we don't, because the state machine macro treats STATE_MACHINE_INITIAL the same as STATE_MACHINE_TLS12.

Reply:
It doesn't matter for using the state machine, but it matters for protecting against switching state machines.